### PR TITLE
GH Actions: use NPM version for docker tag

### PR DESCRIPTION
**Pull request by pdiakumis**
@alexiswl I edited your GH Action workflow to tag the docker image based on the version in `package.json`. The `bump` script can be used to bump the version locally with e.g. `./bump minor`, then the resulting commit will trigger the GH Action upon commit to main or v2.

If you can figure out how to pass the `${{ steps.package-version.outputs.current-version }}` into an env var to pass along through the rest of the workflow (instead of repeating that) it would be perfect ;-)

Also note those other files are from the current umccr/illumination main branch (basically just adding the option to use a `.env` file to keep -untracked- env vars to pass into the app), so they should get resolved I think automatically in your current PR.

### CODE REVIEW

**Comment by pdiakumis** [comment]
Also note that in order to pull the image I needed to do `docker login ghcr.io --username <OWNER> --password <GH_PAT>`, where my `GH_PAT` had read/write permissions for packages.

**Comment by alexiswl** [comment]
> Also note that in order to pull the image I needed to do docker login ghcr.io --username <OWNER> --password <GH_PAT>, where my GH_PAT had read/write permissions for packages.

Shouldn't need this once the package is public.  I can't make it public since it's a fork.

File: .github/workflows/build_docker_container_and_push_to_ghcrio.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - v2
+jobs:
+  tag_and_dockerise:
+    # When merging to one of the branches above and the commit message matches
+    if: "startsWith(github.event.head_commit.message, 'Bump version:')"
+    name: Tag and dockerise
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
```
Comment (alexiswl) [inline .github/workflows/build_docker_container_and_push_to_ghcrio.yml:17]: How does this invoke the bump script?

Comment (pdiakumis) [reply .github/workflows/build_docker_container_and_push_to_ghcrio.yml:17]: That's just a default, the `./bump` script is invoked locally, that bumps the version in `package.json` and `package-lock.json` and commits with the message in `bump`. Then when you push the version gets read and the image gets tagged based on that.

**Comment by alexiswl** [comment]
> If you can figure out how to pass the `${{ steps.package-version.outputs.current-version }}` into an env var to pass along through the rest of the workflow (instead of repeating that) it would be perfect ;-)

We can have it as a separate job, and then it becomes `${{ needs.get_package_version.output.current_version }}` but that's just as long.  There's a few points where it could be substituted in directly instead of being set as the environment variable and then the environment variable placed in, but the env method might be easier to read.  

We could use action.yml and workflow.yml inside folders to generate local methods which then use inputs (similar to a tool in CWL or WDL) but I've had issues with these in the past.

**Comment by pdiakumis** [comment]
Maybe just keep it simple as-is ;-)
That `npm-get-version` action simply does the below, btw:
https://github.com/martinbeentjes/npm-get-version-action/blob/master/entrypoint.sh#L4

File: .github/workflows/build_docker_container_and_push_to_ghcrio.yml (hunk 1 above)
Comment (alexiswl) [reply .github/workflows/build_docker_container_and_push_to_ghcrio.yml:17]: > That's just a default

Ah that makes sense

**Review by alexiswl — APPROVED** [review]

**Pull request merged**